internal/utils/html_fetcher_test.go

```diff
@@ -0,0 +1,106 @@
+package utils
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/PuerkitoBio/goquery"
+	log "github.com/sirupsen/logrus"
+)
+
+func ExampleGetHTML() {
+	var doc interface{}
+
+	doc, err := GetHTML("https://www.google.com/")
+	if err != nil {
+		fmt.Println("Document could not be retrieved")
+	}
+
+	_, ok := doc.(*goquery.Document)
+	if ok {
+		fmt.Println("HTML Document successfully returned")
+	} else {
+		fmt.Println("HTML Document could not be parsed")
+	}
+	// Output: HTML Document successfully returned
+}
+
+func MockServer(testCase string) *httptest.Server {
```

Comment:
```suggestion
func mockServer(testCase string) *httptest.Server {
```

I don't see why you need to export this (e.g. UpperCase).